test/java/seedu/addressbook/logic/LogicTest.java

```diff
@@ -80,6 +80,9 @@ private void assertCommandBehavior(String inputCommand,
         //Execute the command
         CommandResult r = logic.execute(inputCommand);
 
+        //Force to save no matter command is mutating or not
+        saveFile.save(addressBook);
```

Comment:
You need to check for both cases and also if both of them works correctly or not.